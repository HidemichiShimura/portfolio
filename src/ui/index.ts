// Atoms
export { default as PageHeading } from "./atoms/PageHeading/PageHeading";
export { default as Avatar } from "./atoms/Avatar/Avatar";
export { default as Typography } from "./atoms/Typography/Typography";
export { default as SectionHeading } from "./atoms/SectionHeading/SectionHeading";
export { default as Icon } from "./atoms/Icon/Icon";
export { default as RadialLinearContainer } from "./atoms/RadialLinearContainer/RadialLinearContainer";
export { default as Grid } from "./atoms/Grid/Grid";
export { default as GradientCard } from "./atoms/GradientCard/GradientCard";
export { default as Card } from "./atoms/Card/Card";

// Molecules
export { default as HeroHeadings } from "./molecules/HeroHeadings/HeroHeadings";
