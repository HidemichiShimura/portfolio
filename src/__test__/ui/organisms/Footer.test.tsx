import { Footer } from "@/ui";
import { render, screen } from "@testing-library/react";

const EMAIL = "hidemichi.shimura@gmail.com";
const LINKEDIN = "https://www.linkedin.com/in/hidemichi-shimura";
const COPYRIGHT_TEXT = "Â© 2024 Hidemimchi Shimura";

describe("Organisms: <Footer />", () => {
  test("renders", () => {
    render(<Footer />);

    expect(screen.queryByRole("contentinfo")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    render(<Footer />);

    expect(screen.queryByRole("contentinfo")).toMatchSnapshot();
  });

  test("has email and linkedin links", () => {
    render(<Footer />);

    const links = screen.queryAllByRole("link");

    links.forEach((link) => {
      const href = link.getAttribute("href");

      if (href?.startsWith("mailto:")) {
        expect(href).toBe(`mailto:${EMAIL}`);
      } else {
        expect(href).toBe(LINKEDIN);
      }
    });
  });

  test("has copyright notice", () => {
    render(<Footer />);

    expect(screen.queryByText(COPYRIGHT_TEXT)).toBeInTheDocument();
  });
});
