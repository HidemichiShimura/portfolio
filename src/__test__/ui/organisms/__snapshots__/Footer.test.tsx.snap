// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms: <Footer /> matches snapshot 1`] = `
<footer>
  <div
    class="flex h-20 w-full items-center justify-evenly md:h-40"
  >
    <ul
      class="flex gap-2"
    >
      <li
        class="w-7"
      >
        <a
          href="mailto:hidemichi.shimura@gmail.com"
          target="_blank"
        >
          <img
            class="invert-half"
            src="/img.jpg"
          />
        </a>
      </li>
      <li
        class="w-7"
      >
        <a
          href="https://www.linkedin.com/in/hidemichi-shimura"
          target="_blank"
        >
          <img
            class="invert-half"
            src="/img.jpg"
          />
        </a>
      </li>
    </ul>
    <p
      class="text-light-gray text-base font-sans"
    >
      Â© 
      2024
       Hidemimchi Shimura
    </p>
  </div>
</footer>
`;
